hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/server/protocol/StorageReport.java

```diff
@@ -48,6 +49,8 @@ public StorageReport(DatanodeStorage storage, boolean failed, long capacity,
     this.nonDfsUsed = nonDfsUsed;
     this.remaining = remaining;
     this.blockPoolUsed = bpUsed;
+    this.blockPoolUsagePercent = capacity == 0 ? 0.0f :
```

Comment:
If I remember right, `capacity` can be 0.
```suggestion
    this.blockPoolUsagePercent = capacity <= 0 ? 0.0f :
```

Reply:
Thanks @tasanuma for your review. Thus can prevent some anomalies. I will update it soon.